### PR TITLE
Update/add resources with support for embedded resources for SPDX and/or custom policies

### PR DESCRIPTION
**Pull request by markstur**
* Allow --addAll to work with custom or spdx (embedded) or customPath or spdxPath (not embedded)
* Implemented --updateAll which is like --addAll but for in-place update (i.e., update prechecks)
* Read/Write code is going through Resources
  - Support for embedded resources
  - More code re-use
  - More encapsulation of file naming and directory structure
* Removed --addPattern because it was never implemented (we have --addAll)

Embedded resources help the API/build use case!

* --spdx and --custom refer only to the embedded dirs under our resources dir
* To use other external resources, specify --customPath or --spdxPath
* Stopped using config entry "resources" in favor of separate --customPath and --spdxPath
  - "resources" was handy for tests and things but is confusing to document. Our primary use case is probably embedded --spdx plus custom/customPath policies that are either embedded or not.
  - Default config.json is no longer needed (was just resources="resources") but config files can still be used to override default flags.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -112,14 +112,14 @@ $ license-scanner --help
 Usage:
   license-scanner [flags]
 
-
 Flags:
   -g, --acceptable          Flag acceptable
-      --addAll string       Add the licenses from SPDX unzipped release
+      --addAll string       Add licenses
```
Comment (pritidesai) [inline README.md:117]: thanks @markstur 👍 

I think we might want to add a little bit more description here similar to what we have in the PR description:

```
--addAll to work with custom or spdx (embedded) or customPath or spdxPath (not embedded)
```

Or what exactly this option does, addAll can be set to a path with the SPDX released templates or custom licenses.

File: README.md [hunk 2]
```diff
@@ -517,16 +528,23 @@ ok      github.com/CycloneDX/license-scanner/normalizer 0.161s
...
+
+Imported custom policies (and generated precheck files) will be copied into a destination specified by the --customPath or --custom argument.
+
+* `--custom <subDir>`: The destination must be an empty or non-existent directory under `resources/custom/`. These policies will be available as embedded resources when you scan for licenses using the same `--custom <subDir>` flag.
+
+* `--customPath <path>`: The destination must be an empty or non-existent directory. These copied policies and precheck files will not be available as embedded resources, but can be used when you scan for licenses using the `--customPath <path>` flag to read this external directory.
+
+When importing, only one of `--spdx, --spdxPath, --custom, --customPath` can be used. This will be the destination for the file files copied from the `--addAll <directory>`. When scanning for licenses, you can combine one set of SPDX templates specified by `--spdx` or `--spdxPath` with one set of custom policies specified by `--custom` or `--customPath`.
```
Comment (pritidesai) [inline README.md:547]: NIT: extra file in for the file files copied from the ...

File: api/scanner/scan_resources_test.go [hunk 3]
```diff
@@ -62,21 +66,25 @@ func TestScanner_Resources(t *testing.T) {
...
 				name := tt.name + " with SPDX=" + spdx + " Custom=" + custom
 				t.Run(name, func(t *testing.T) {
+					t.Parallel()
 					flagSet := configurer.NewDefaultFlags()
-					_ = flagSet.Set("configPath", "../../testdata/resources")
+					_ = flagSet.Set("configPath", path.Join("..", "..", "testdata", "resources"))
 					if spdx != doNotSet {
-						_ = flagSet.Set(configurer.SpdxFlag, spdx)
+						_ = flagSet.Set("spdxPath", path.Join("..", "..", "testdata", "resources", "spdx", spdx))
+						_ = flagSet.Set("spdxPath", path.Join("..", "..", "testdata", "resources", "spdx", spdx))
```
Comment (pritidesai) [inline api/scanner/scan_resources_test.go:84]: these settings are identical, is it on purpose?

File: cmd/license-scanner.md [hunk 4]
```diff
@@ -44,7 +44,9 @@ license-scanner [flags]
...
   -n, --normalized          Flag normalized
   -q, --quiet               Set logging to quiet
-      --spdx string         SPDX templates to use (default "default")
+      --spdx string         Set of embedded SPDX templates to use (default "default")
+      --spdxPath string     Path to external SPDX templates to use
+      --updateAll           Update existing licenses
 ```
 
-###### Auto generated by spf13/cobra on 6-Oct-2022
+###### Auto generated by spf13/cobra on 23-Nov-2022
```
Comment (pritidesai) [inline cmd/license-scanner.md:52]: NIT: might want to update this to the latest? 🤔

File: cmd/root_test.go [hunk 5]
```diff
@@ -135,29 +139,59 @@ func Test_CLI_addAll(t *testing.T) {
...
+		"--configPath", "../testdata/config",
 		"--configName", "bogus",
+		"-f", "../testdata/addAll/input/text/0BSD.txt",
 	})
 	if err := cmd.Execute(); !errors.As(err, &viper.ConfigFileNotFoundError{}) {
 		t.Fatalf("Expected ConfigFileNotFoundError got: %v", err)
 	}
 }
 
+// Test_CLI_default_configName_not_found_is_ok verifies that it is ok to NOT have a config.* file with configPath set
```
Comment (pritidesai) [inline cmd/root_test.go:180]: thank you @markstur for adding a comprehensive list of tests, appreciate it 👍

**Review by pritidesai — APPROVED** [review]

**Comment by pritidesai** [comment]
thanks a bunch @markstur for the cleanup and adding support for both SPDX and custom licenses 👍

**Comment by pritidesai** [comment]
the changes look good to me, I was trying to double confirm the unit tests all passing but couldn't find that check 🙃

**Comment by markstur** [comment]
> the changes look good to me, I was trying to double confirm the unit tests all passing but couldn't find that check 🙃

In case anyone else sees this... the test action was fixed in a separate PR #15

**Pull request merged**